javalib/src/main/scala/java/util/TreeSet.scala

```diff
@@ -142,7 +142,6 @@ class TreeSet[E] (_comparator: Comparator[_ >: E])
     val boxedFrom = Box(fromElement)
     val boxedTo = Box(toElement)
     val subSetFun = { () =>
-      // the creation of a new TreeSet is to avoid a mysterious bug with scala 2.10
```

Comment:
Hmmm... What happens if you set `var base = inner.range(boxedFrom, boxedTo)`?

Reply:
`range` does not seem to guarantee that it returns a fresh `TreeSet`, independent of `inner`. So in theory it is possible that we receive something that will be mutated under our feet if we do not force the copy into a new `TreeSet`.

I haven't observed the behavior, but it could happen in future versions of the Scala lib in any case.